### PR TITLE
Feature/use GitHub docker registry

### PR DESCRIPTION
**Pull request by darren-rose**
### Motivation

publishing to dockerhub now requires pro account

### Modification

use github container registry 

@thomaseizinger @bonomat

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    # Publish semver tags as releases.
+    tags: [ 'v*.*.*' ]
+  pull_request:
+    branches: [ main, master ]
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
```
Comment (thomaseizinger) [inline .github/workflows/docker-image.yml]: Our image name doesn't follow this naming pattern.

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  push:
+    branches: [ main, master ]
+    # Publish semver tags as releases.
+    tags: [ 'v*.*.*' ]
+  pull_request:
+    branches: [ main, master ]
```
Comment (thomaseizinger) [inline .github/workflows/docker-image.yml]: I don't think we need to run on PRs at all. Then we can remove the conditonal steps from below.

File: .github/workflows/docker-image.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # Login against a Docker registry except on PR
+      # https://github.com/docker/login-action
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@28218f9b04b4f3f62068d7b6ce6ca5b26e35336c
```
Comment (thomaseizinger) [inline .github/workflows/docker-image.yml]: What is the reason for depending on a specific hash here? Is there something wrong with the latest version (1.10)?

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (thomaseizinger) [reply .github/workflows/docker-image.yml]: We are only using this variable once, why the indirection?

File: .github/workflows/docker-image.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+      # Build and push Docker image with Buildx (don't push on PR)
+      # https://github.com/docker/build-push-action
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```
Comment (thomaseizinger) [inline .github/workflows/docker-image.yml]: I've never used this before, how does it work exactly? `master` will map to `latest` and releases will map to individual tags?

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (thomaseizinger) [reply .github/workflows/docker-image.yml]: I guess we could rename it to make things easier.

**Review by thomaseizinger — COMMENTED** [review]
Thanks for taking on this effort!

Couple of questions in the comments.

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (darren-rose) [reply .github/workflows/docker-image.yml]: I will remove it but the docker images need to follow the github repo naming

File: .github/workflows/docker-image.yml (hunk 2 above)
Comment (darren-rose) [reply .github/workflows/docker-image.yml]: will remove

File: .github/workflows/docker-image.yml (hunk 3 above)
Comment (darren-rose) [reply .github/workflows/docker-image.yml]: will resolve

File: .github/workflows/docker-image.yml (hunk 4 above)
Comment (darren-rose) [reply .github/workflows/docker-image.yml]: used to tag and label docker images

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (thomaseizinger) [reply .github/workflows/docker-image.yml]: Didn't know that, thanks for pointing it out!

**Review by thomaseizinger — APPROVED** [review]
Thanks!

Just renamed the repository, lets see if it works!

**Pull request merged**